File: 28-find-the-index-of-the-first-occurrence-in-a-string/28-find-the-index-of-the-first-occurrence-in-a-string.rb
# @param {String} haystack
# @param {String} needle
# @return {Integer}
def str_str(haystack, needle)
    n_len = needle.size
    h_len = haystack.size
    
    (0..(h_len - n_len)).each do |i|
        return i if haystack[i...i+n_len] == needle
    end
    
    -1
end

File: 28-find-the-index-of-the-first-occurrence-in-a-string/28-find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    const len = needle.length
    for(let i = 0; i <= haystack.length - len; i++) {
        if(haystack.slice(i, i+len) === needle) return i;
    }
    return -1;
};